gr-qtgui/examples/c++/Number_Sink.grc compilation error

### What happened?

Starting flowgraph gr-qtgui/examples/c++/Number_Sink.grc throws a compilation error and will not run to completion.

### System Information

OS: Ubuntu 24.04
GR Installation Method: source


### GNU Radio Version

3.11-git (main)

### Specific Version

v3.11.0.0git-816-g30a0aa91

### Steps to Reproduce the Problem

Run the flowgraph

### Relevant log output

```shell
Generating: "/home/matt/gnuradio/gr-qtgui/examples/c++/Number_Sink"
>>> Warning: The block 'blocks_throttle_0' is deprecated.

Executing: cmake .. && cmake --build . -j3 && cd ../.. && /usr/bin/x-terminal-emulator -e /home/matt/gnuradio/gr-qtgui/examples/c++/Number_Sink/build/Number_Sink

-- Using GMP.
-- User set python executable /usr/bin/python3
CMake Warning (dev) at /usr/local/lib/cmake/gnuradio/GrPython.cmake:21 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/local/lib/cmake/gnuradio/GnuradioConfig.cmake:37 (include)
  CMakeLists.txt:12 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/local/lib/cmake/gnuradio/GrPython.cmake:27 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/local/lib/cmake/gnuradio/GnuradioConfig.cmake:37 (include)
  CMakeLists.txt:12 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- QWT Version: 6.1.4
-- Build type not specified: defaulting to release.
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /home/matt/gnuradio/gr-qtgui/examples/c++/Number_Sink/build
[  0%] Built target Number_Sink_autogen_timestamp_deps
[ 20%] Automatic MOC for target Number_Sink
[ 20%] Built target Number_Sink_autogen
[ 40%] Building CXX object CMakeFiles/Number_Sink.dir/Number_Sink.cpp.o
/home/matt/gnuradio/gr-qtgui/examples/c++/Number_Sink/Number_Sink.cpp:115:6: error: redefinition of ‘void Number_Sink::closeEvent(QCloseEvent*)’
  115 | void Number_Sink::closeEvent(QCloseEvent *event) {
      |      ^~~~~~~~~~~
/home/matt/gnuradio/gr-qtgui/examples/c++/Number_Sink/Number_Sink.cpp:100:6: note: ‘virtual void Number_Sink::closeEvent(QCloseEvent*)’ previously defined here
  100 | void Number_Sink::closeEvent(QCloseEvent *event) {
      |      ^~~~~~~~~~~
gmake[2]: *** [CMakeFiles/Number_Sink.dir/build.make:96: CMakeFiles/Number_Sink.dir/Number_Sink.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:86: CMakeFiles/Number_Sink.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

>>> Done (return code 2)
```

--- a/gr-qtgui/grc/qtgui_sink_x.block.yml
+++ b/gr-qtgui/grc/qtgui_sink_x.block.yml
@@ -138,7 +138,7 @@ templates:
 
 
 cpp_templates:
-  includes: ['#include <gnuradio/qtgui/${type.fcn}.h>']
+  includes: ['#include <gnuradio/qtgui/${type.fcn}.h>', '#include <gnuradio/fft/window.h>']
   declarations: gr::qtgui::${type.fcn}::sptr ${id};
   make: |-
     <%

--- a/grc/core/generator/cpp_templates/flow_graph.cpp.mako
+++ b/grc/core/generator/cpp_templates/flow_graph.cpp.mako
@@ -102,10 +102,6 @@ ${var.vtype} ${class_name}::get_${var.name} () const {
     return this->${var.name};
 }
 
-void ${class_name}::closeEvent(QCloseEvent *event) {
-this->settings->setValue("geometry",this->saveGeometry());
-event->accept();
-}
 
 void ${class_name}::set_${var.name} (${var.vtype} ${var.name}) {
 % if flow_graph.get_option('thread_safe_setters'):
@@ -120,6 +116,13 @@ void ${class_name}::set_${var.name} (${var.vtype} ${var.name}) {
 }
 
 % endfor
+
+% if generate_options == 'qt_gui':
+    void ${class_name}::closeEvent(QCloseEvent *event) {
+    this->settings->setValue("geometry",this->saveGeometry());
+    event->accept();
+}
+% endif
 
 int main (int argc, char **argv) {
     % if parameters:

